src/mscorlib/src/System/Threading/SpinLock.cs

```diff
@@ -337,32 +337,28 @@ private void ContinueTryEnter(int millisecondsTimeout, ref bool lockTaken)
                 Thread.BeginCriticalRegion();
 #endif
 
-                if (Interlocked.CompareExchange(ref m_owner, observedOwner | 1, observedOwner, ref lockTaken) == observedOwner)
+                if (Interlocked.CompareExchange(ref m_owner, observedOwner | 1, observedOwner, ref lockTaken) == observedOwner ||
+                    millisecondsTimeout == 0)
```

Comment:
The `millisecondsTimeout == 0` check should be after EndCriticalRegion - it will break the full framework build otherwise.